agent/rpc/peering/stream_test.go

```diff
@@ -63,10 +66,32 @@ func TestStreamResources_Server_Follower(t *testing.T) {
 		}
 	}()
 
+	// expect error
 	msg, err := client.Recv()
 	require.Nil(t, msg)
 	require.Error(t, err)
 	require.EqualError(t, err, "rpc error: code = FailedPrecondition desc = cannot establish a peering stream on a follower node")
+
+	// expect a status error
+	st, ok := status.FromError(err)
+	require.True(t, ok, "need to get back a grpc status error")
+	deets := st.Details()
+	// expect a LeaderAddress message
+	{
+		found := false
+		for _, deet := range deets {
+			la, ok := deet.(*pbpeering.LeaderAddress)
+
+			if !ok {
+				continue
+			} else {
+				found = true
+				require.Equal(t, "expected:address", la.Address)
+			}
+		}
+
+		require.True(t, found, "expected to find a LeaderAddress message in error details")
+	}
```

Comment:
Can we simplify this with `require.ElementsMatch`? Unsure if it works but this is quite verbose for what should be a simple slice assertion

Reply:
Chatting offline, I don't think it's possible to use `require.ElementsMatch` or `require.Contains` since `deets` contains `interface{}` types that need to be typecast. If we went to typecast, then we might as well look for the element we want to find like in the closure above.

Reply:
It's actually not about the typecast; you can do something like:
```go
	type Blah struct {
		Name string
	}
	expect := []Blah{{Name: "hi"}}
	got := []interface{}{Blah{Name: "hi"}}
	require.ElementsMatch(t, expect, got)
```

The reason it doesn't work in our case is because proto structs [have internal state](https://developers.google.com/protocol-buffers/docs/reference/go/faq#deepequal).

But there are solutions!
```go
	deets := st.Details()

	exp := []interface{}{
		&pbpeering.LeaderAddress{
			Address: "expected:address",
		},
	}
	prototest.AssertDeepEqual(t, exp, deets)
```